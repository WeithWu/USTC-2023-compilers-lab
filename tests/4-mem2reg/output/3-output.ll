; ModuleID = 'cminus'
source_filename = "/home/weith/gits/USTC-2023-compilers-lab/tests/4-mem2reg/functional-cases/3-output.cminus"

declare i32 @input()

declare void @output(i32)

declare void @outputFloat(float)

declare void @neg_idx_except()

define i32 @main() {
label_entry0:
  call void @output(i32 11)
  call void @output(i32 22222)
  ret i32 0
}
