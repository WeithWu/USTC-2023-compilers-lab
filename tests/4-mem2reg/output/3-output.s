	.text
	.globl main
	.type main, @function
main:
	st.d $ra, $sp, -8
	st.d $fp, $sp, -16
	addi.d $fp, $sp, 0
	addi.d $sp, $sp, -16
.main_label_entry0:
# call void @output(i32 11)
	addi.w $a0, $zero, 11
	bl output
# call void @output(i32 22222)
	lu12i.w $a0, 5
	ori $a0, $a0, 1742
	bl output
# ret i32 0
	addi.w $a0, $zero, 0
	addi.d $sp, $sp, 16
	ld.d $ra, $sp, -8
	ld.d $fp, $sp, -16
	jr $ra
	addi.d $sp, $sp, 16
	ld.d $ra, $sp, -8
	ld.d $fp, $sp, -16
	jr $ra
