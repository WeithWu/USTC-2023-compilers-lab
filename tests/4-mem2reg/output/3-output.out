11
22222
0
